plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcClient.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.client;
+
+import com.google.common.net.HostAndPort;
+import io.trino.plugin.pinot.PinotConfig;
+import org.apache.pinot.common.proto.Server;
+import org.apache.pinot.common.request.BrokerRequest;
+import org.apache.pinot.common.utils.grpc.GrpcQueryClient;
+import org.apache.pinot.common.utils.grpc.GrpcRequestBuilder;
+import org.apache.pinot.sql.parsers.CalciteSqlCompiler;
+
+import javax.inject.Inject;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.util.Objects.requireNonNull;
+
+public class PinotGrpcClient
+{
+    private static final CalciteSqlCompiler REQUEST_COMPILER = new CalciteSqlCompiler();
+
+    private final PinotHostMapper pinotHostMapper;
+    private final Map<HostAndPort, GrpcQueryClient> clientCache = new ConcurrentHashMap<>();
+    private final int grpcPort;
+
+    @Inject
+    public PinotGrpcClient(PinotConfig pinotConfig, PinotHostMapper pinotHostMapper)
+    {
+        this.pinotHostMapper = requireNonNull(pinotHostMapper, "pinotHostMapper is null");
+        this.grpcPort = requireNonNull(pinotConfig, "pinotConfig is null").getGrpcPort();
+    }
+
+    public Iterator<Server.ServerResponse> submitQuery(String query, String serverHost, List<String> segments)
+    {
+        HostAndPort mappedHostAndPort = pinotHostMapper.getServerGrpcHostAndPort(serverHost, grpcPort);
+        GrpcQueryClient client = clientCache.computeIfAbsent(mappedHostAndPort, k -> new GrpcQueryClient(mappedHostAndPort.getHost(), mappedHostAndPort.getPort()));
```

Comment:
The lambda should use `k`. Also, give it a more meaninful name. At a minimum, `key`, but maybe `hostAndPort` would make it more obvious and readable.

Reply:
Who manages the lifecycle of these clients? Do they need to be torn down? Can they go into an "invalid" state (in which case, they'd need to be refreshed)?

Reply:
Grpc clients are long lived and use internal threads to manage connections.
TODO:
* Manage lifecycle - close grpc clients on shutdown
* Separate configs and modules for grpc and legacy query clients
* Remove unused configs (separate commit)
* TBD: extract to separate commits